src/kernel-install/kernel-install

```diff
@@ -139,6 +143,18 @@ case $COMMAND in
             exit 1
         fi
 
+        # Compatibility with earlier versions that used the presence of $BOOT_ROOT/$MACHINE_ID
+        # to signal to 00-entry-directory to create $ENTRY_DIR_ABS
+        # to serve as the indication to use or to not use the BLS
+        if [ "$KERNEL_INSTALL_BOOTLOADER_SPEC" != 0 ]; then
+            if [ "$KERNEL_INSTALL_VERBOSE" -gt 0 ]; then
+                echo "+mkdir -v -p $ENTRY_DIR_ABS"
+                mkdir -v -p "$ENTRY_DIR_ABS"
+            else
+                mkdir -p "$ENTRY_DIR_ABS"
+            fi
+        fi
```

Comment:
I'd prefer to keep this in `00-entry-directory.install` as before. Especially if it's just compat, it's better not to make `kernel-install` involved in this.

Reply:
Hm, I'm slightly concerned that that'll always leave garbage behind (in the form of `$ENTRY_DIR_ABS`) without a good way to clean it – making 00-.install remove it wholesale would potentially raze other plugins' data in the process.
Maybe run the plugins back-to-front on remove (and facilitate more reasonable unwinding)?

Reply:
If we can get away with it, running back-to-front on remove sounds super reasonable. @keszybz do you think doing this would break compat with existing scripts?